src/Operation.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Aidantwoods\SecureHeaders;
+
+interface Operation
+{
+    /**
+     * Transform the given set of headers
+     *
+     * @param HeaderBag $headers
+     * @return void
+     */
+    public function modify(HeaderBag $headers);
```

Comment:
Seeing the code of other PHP libraries, I'd say the knowledge is very common in PHP land.

Also, passing objects by value could get super expensive with deep cloning etc., so I don't think this is really a problem.